lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -56,6 +56,40 @@ def capture(money, authorization, options = {})
         commit('capture', post)
       end
 
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+      end
+
+      def refund(amount, authorization, options = {})
+        post = {
+          request: {
+            MerchantId: @options[:merchant_id],
+            OrderId: authorization,
+            ReferenceId: options[:reference_id],
+            Amount: amount
+          }
+        }
+
+        commit('refund', post)
+      end
+
+      def void(authorization, options = {})
+        post = { request: { MerchantId: @options[:merchant_id], OrderId: authorization } }
+
+        commit('cancel', post)
+      end
+
+      def verify(credit_card, options = {})
```

Comment:
⚠️ Warning/Suggestion:

Here you have some testable behavior on how you want your reponse, is posible to add a unit test that checks that:
- Response is Multireponse instance
- The authorize response is the default one
- last response is not used